lua/pac3/core/client/parts/light.lua

```diff
@@ -68,6 +83,10 @@ function PART:OnDraw()
 	self:GetLight().pos = pos
 	self:GetLight().dir = ang:Forward()
 end
+function PART:SetStyle(val)
+	self.Style = val
+	self:GetLight().Style = self.Style
```

Comment:
I'm not sure how useful this is. You can mimic a light style by just adding a proxy?

Reply:
Those are a source engine feature and they were in the legacy lights.

No, the styles aren't easily replicated with proxies. They're arbitrary sequences of levels, most of them are not a nice continuous function. Also it's easier to setup something quick that way.

Reply:
> if they were in legacy and not in the new, that means I didn't observe any purpose for them and let them out to reduce the amount of properties. They are also not smoothly played back as far as I can remember, feeling very hl1-like.

Well, I see a purpose. But now we care about the amount of properties? You didn't seem to object when there were other parts I've added more stuff to. I added data sources to the text part, a lot of them are single-liner functions but we want them for ease of setting up.

And I'm one of the proxy enthusiasts, you'd think I'd want more people using proxies. But these are not extra steps we need.

I think people like having easier options that are readily available. And I'm not a data guy but roughly judging from the discord's help channel, most people don't use proxies or know much about them. So I see ample reason to make things easier for users. Anyone with a light can easily get something nice in two clicks.

> Perhaps one idea is a proxy function that allows you to loop a sequence of numbers, a bit like the interpolator.

New proxy function ideas? Now that's right up my alley.